pkg/apiserver/api/v1/instance.proto

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+package kubecarrier.api.v1;
+option go_package = "v1";
+
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+
+import "meta.proto";
+
+message Instance {
+  ObjectMeta metadata = 1;
+  string spec = 2;
+  string status = 3;
+}
+
+message InstanceList {
+  ListMeta metadata = 1;
+  repeated Instance items = 2;
+}
+
+message InstanceGetRequest {
+  string instance = 1;
+  string version = 2;
+  string name = 3;
+  string account = 4;
+}
+
+message InstanceDeleteRequest {
+  string instance = 1;
+  string version = 2;
```

Comment:
What has the version got to do with the deletion request? It shouldn't matter since the version is useful only for get/listing, for deletion is irrelevant.

Reply:
to be consistent with all other methods.
`/v1/accounts/{account}/instance/{instance}/{version}/{name}`
don't want to have empty {version} in that path

Reply:
But this parameter is unnecessary. Maybe reverse the ordering as:
 
`/v1/accounts/{account}/instance/{instance}/{name}/{version}`

for all other API calls?

At the end of the day, I don't really care that much for REST API, since gRPC-web should be used by default.

Reply:
but for the listing we have `"/v1/accounts/{account}/instances/{offering}/{version}"`, so we can't revers that